### PR TITLE
Changed account id structure

### PR DESCRIPTION
**Pull request by Kayanski**
This PR aims at changing the account-id structure to be able to host interchain accounts.
This adds a trace to the account id, to identify the origin of an account.

This trace allows multiple chained chain calls for now. It may be limited to a single chain in the future to only allow one chain hop (no reason to have multiple hops)

### Checklist

- [x] CI is green.
- [x] Changelog updated.

### CODE REVIEW

**Review by Buckram123 — APPROVED** [review]
Nice, can't wait to have remote account.
Cool `ChainName` implementation as well!

File: framework/contracts/native/account-factory/src/commands.rs [hunk 1]
```diff
@@ -51,10 +53,20 @@ pub fn execute_create_account(
     let config = CONFIG.load(deps.storage)?;
 
     let governance = governance.verify(deps.as_ref(), config.version_control_contract.clone())?;
+
+    // We want to create a local account
+    // and set the origin to local
+    let origin = AccountTrace::Local;
```
[the remark this answers is not in the document]
Comment (Kayanski) [reply framework/contracts/native/account-factory/src/commands.rs:59]: Yes exactly : https://github.com/AbstractSDK/abstract/pull/37/files#diff-2cdefa8bf045807cde1a2d1c701cd2120f6f42f5d6b69d4ea935721952e70dcf

File: framework/contracts/native/version-control/src/commands.rs [hunk 2]
```diff
@@ -38,7 +39,13 @@ pub fn add_account(
 ) -> VCResult {
     // Only Factory can add new Account
     FACTORY.assert_admin(deps.as_ref(), &msg_info.sender)?;
-    ACCOUNT_ADDRESSES.save(deps.storage, account_id, &account_base)?;
+    // Check if account already exists
+    ensure!(
```
Comment (CyberHoward) [inline framework/contracts/native/version-control/src/commands.rs:43]: nice!

Comment (Kayanski) [reply framework/contracts/native/version-control/src/commands.rs:43]: You did that if I remember correctly

Comment (CyberHoward) [reply framework/contracts/native/version-control/src/commands.rs:43]: Extra nice ;)

**Pull request merged**